### ISSUE TITLE
Using without models from insightface?

### ISSUE THREAD

**Issue by JohnMe** [foivospar/Arc2Face#1]
Hi,
is there any way to use this without downloading the models from insightface, since they have a really restrictive license...

It seems your example is only using the 2 small files:

2d106det.onnx
scrfd_10g_bnkps.onnx

Is it possible to pass an image directly into 
arcface.onnx
if i already detected the face part?

**Comment by jadechip** [comment]
+1

**Comment by jadechip** [comment]
Also curious if you are planning on releasing the training code? Thank you.

**Comment by foivospar** [comment]
Please see https://github.com/foivospar/Arc2Face/issues/3#issuecomment-2023165300.

**Comment by jadechip** [comment]
@foivospar could you kindly elaborate a bit on how insightface is being used, and if it's non-commercial license applies to arcface? From my understanding of the antelope package,
scrfd_10g_bnkps.onnx is used for initial face detection.
2d106det.onnx provides detailed facial keypoints.
glintr100.onnx is responsible for generating face embeddings.
...does arcface replace these?

**Comment by foivospar** [comment]
No, we only replace the face recognition module, i.e. `glintr100.onnx`, with `arcface.onnx`. The rest are downloaded from InsightFace, so their license applies. By the way, only the detection module `scrfd_10g_bnkps.onnx` is needed, as we do not use keypoints, gender annotations, etc.

**Comment by jadechip** [comment]
Thank you @foivospar. I am looking into training a non-commercial version of the insightface models, including scrfd_10g_bnkps.onnx. Will report back if I make any progress.

**Comment by fofr** [comment]
@jadechip Have you made any progress on your non-commercial version?

**Comment by pevny** [comment]
> No, we only replace the face recognition module, i.e. `glintr100.onnx`, with `arcface.onnx`. The rest are downloaded from InsightFace, so their license applies. By the way, only the detection module `scrfd_10g_bnkps.onnx` is needed, as we do not use keypoints, gender annotations, etc.

Hello @foivospar 
does it mean that if one would get the bounding boxes with other face detection model than `scrfd_10g_bnkps.onnx` we would be insightface free?
